topo/node/node.go

```diff
@@ -72,91 +78,43 @@ func Register(t topopb.Node_Type, fn NewNodeFn) {
 	mu.Unlock()
 }
 
-// Node is a topology node in the cluster.
-type Node struct {
-	impl       Implementation
-	namespace  string
-	kClient    kubernetes.Interface
-	rCfg       *rest.Config
-	interfaces map[string]*Link
-}
-
-// Interfaces returns the node's map of interfaces.
-func (n *Node) Interfaces() map[string]*Link {
-	return n.interfaces
-}
-
-// KubeClient returns the node's kubeclient.
-func (n *Node) KubeClient() kubernetes.Interface {
-	return n.kClient
-}
-
-// RESTConfig returns the node's REST configuration.
-func (n *Node) RESTConfig() *rest.Config {
-	return n.rCfg
-}
-
-// Namespace returns the node's namespace.
-func (n *Node) Namespace() string {
-	return n.namespace
+// Vendor registers the vendor type with the topology manager.
+func Vendor(v tpb.Vendor, fn NewNodeFn) {
+	mu.Lock()
+	if _, ok := vendorTypes[v]; ok {
+		panic(fmt.Sprintf("duplicate registration for %T", v))
+	}
+	vendorTypes[v] = fn
+	mu.Unlock()
 }
 
-// Impl returns the node implementation.
-func (n *Node) Impl() Implementation {
-	return n.impl
+// Impl is a topology node in the cluster.
+type Impl struct {
+	Namespace  string
+	KubeClient kubernetes.Interface
+	RestConfig *rest.Config
+	Proto      *tpb.Node
+	BasePath   string
 }
 
 // New creates a new node for use in the k8s cluster.  Configure will push the node to
 // the cluster.
-func New(namespace string, pb *topopb.Node, kClient kubernetes.Interface, rCfg *rest.Config) (*Node, error) {
-	impl, err := getImpl(pb)
-	if err != nil {
-		return nil, err
-	}
-	return &Node{
-		namespace:  namespace,
-		impl:       impl,
-		rCfg:       rCfg,
-		kClient:    kClient,
-		interfaces: map[string]*Link{},
-	}, nil
+func New(namespace string, pb *tpb.Node, kClient kubernetes.Interface, rCfg *rest.Config, bp string) (Node, error) {
+	return getImpl(&Impl{
+		Namespace:  namespace,
+		Proto:      pb,
+		KubeClient: kClient,
+		RestConfig: rCfg,
+		BasePath:   bp,
+	})
 }
 
-// Configure creates the node on the k8s cluster.
-func (n *Node) Configure(ctx context.Context, bp string) error {
-	pb := n.impl.Proto()
-	var data []byte
-	switch v := pb.Config.GetConfigData().(type) {
-	case *topopb.Config_File:
-		var err error
-		data, err = ioutil.ReadFile(filepath.Join(bp, v.File))
-		if err != nil {
-			return err
-		}
-	case *topopb.Config_Data:
-		data = v.Data
-	}
-	if data != nil {
-		cm := &corev1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: fmt.Sprintf("%s-config", pb.Name),
-			},
-			Data: map[string]string{
-				pb.Config.ConfigFile: string(data),
-			},
-		}
-		sCM, err := n.kClient.CoreV1().ConfigMaps(n.namespace).Create(ctx, cm, metav1.CreateOptions{})
-		if err != nil {
-			return err
-		}
-		log.Infof("Server Config Map:\n%v\n", sCM)
-	}
-	return nil
+func (n *Impl) GetProto() *tpb.Node {
```

Comment:
would love to but it causes all kinds of headaches since the Impl has an attribute Proto
so when around and around but it was only more confusing on how to deal with that